src/expandableCalendar/WeekCalendar/presenter.js

```diff
@@ -0,0 +1,187 @@
+import _ from 'lodash';
+import React from 'react';
+import {sameWeek} from '../../dateutils';
+const commons = require('../commons');
+import XDate from 'xdate';
+
+const UPDATE_SOURCES = commons.UPDATE_SOURCES;
+// must be a positive number
+const NUMBER_OF_PAGES = 2;
+
+class Presenter {
+  constructor(props) {
+    this.list = React.createRef();
+    this.props = props;
+    this._applyAndroidRtlFix = commons.isAndroid && commons.isRTL;
+    // On Android+RTL there's an initial scroll that cause issues
+    this._firstAndroidRTLScrollIgnored = !this._applyAndroidRtlFix;
+  }
+
+  updateWith = props => {
+    this.props = props;
+  };
+
+  scrollToIndex = animated => {
+    this.list.current.scrollToIndex({animated, index: NUMBER_OF_PAGES});
+  };
+
+  isSameWeek = (date, prevDate, firstDay) => {
+    return sameWeek(date, prevDate, firstDay);
+  };
+
+  // Events
+
+  onDayPressed = (context, value) => {
+    _.invoke(context, 'setDate', value.dateString, UPDATE_SOURCES.DAY_PRESS);
+  };
+
+  onScroll = ({context, updateState, x, page, items, width}) => {
+    if (!this._firstAndroidRTLScrollIgnored) {
+      this._firstAndroidRTLScrollIgnored = true;
+      return;
+    }
+
+    x = this._getX(x, items?.length, width);
+    const newPage = this._getNewPage(x, width);
+
+    if (this._shouldUpdateState(page)) {
+      this._invokeSetDate(context, items[newPage]);
+      const data = this._getItemsForPage(page, items);
+      updateState(data, newPage);
+    }
+  };
+
+  onMomentumScrollEnd = ({items, props, page, updateState}) => {
+    if (this._isFirstPage(page) || this._isLastPage(page, items)) {
+      this.scrollToIndex(false);
+
+      const newWeekArray = this.getDatesArray(props);
+      let updatedItems;
+      if (this._isLastPage(page, items)) {
+        updatedItems = this._mergeArraysFromEnd(items, newWeekArray);
+      } else {
+        updatedItems = this._mergeArraysFromTop(items, newWeekArray);
+      }
+
+      updateState(updatedItems);
+    }
+  };
+
+  shouldComponentUpdate = (context, prevContext, firstDay) => {
```

Comment:
This is a life cycle method. Don't we want it in the component file for readability?

Reply:
On second thought, forget it. The tests are more important in this case.